workflow/src/main/java/com/synopsys/integration/alert/workflow/message/NotificationReceiver.java

```diff
@@ -51,32 +54,46 @@
 
     @Autowired
     public NotificationReceiver(Gson gson, NotificationAccessor notificationAccessor, NotificationProcessor notificationProcessor, ChannelEventManager eventManager) {
-        super(gson, NotificationEvent.class);
+        super(gson, NotificationReceivedEvent.class);
         this.notificationAccessor = notificationAccessor;
         this.notificationProcessor = notificationProcessor;
         this.eventManager = eventManager;
     }
 
     @Override
-    public void handleEvent(NotificationEvent event) {
-        if (NotificationEvent.NOTIFICATION_EVENT_TYPE.equals(event.getDestination())) {
-            if (null == event.getNotificationIds() || event.getNotificationIds().isEmpty()) {
-                logger.warn("Can not process a notification event without notification Id's.");
-                return;
-            }
-            logger.debug("Event {}", event);
-            logger.info("Processing event for {} notifications.", event.getNotificationIds().size());
-            List<AlertNotificationModel> notifications = notificationAccessor.findByIds(event.getNotificationIds());
+    public void handleEvent(NotificationReceivedEvent event) {
+        if (!NotificationReceivedEvent.NOTIFICATION_RECEIVED_EVENT_TYPE.equals(event.getDestination())) {
+            logger.warn("Received an event of type '{}', but this listener is for type '{}'.", event.getDestination(), NotificationReceivedEvent.NOTIFICATION_RECEIVED_EVENT_TYPE);
+            return;
+        }
+        logger.debug("Event {}", event);
+        logger.info("Processing event for notifications.");
+
+        int numPagesProcessed = 0;
+        int pageSize = 100;
+
+        AlertPagedModel<AlertNotificationModel> pageOfAlertNotificationModels = notificationAccessor.getFirstPageOfNotificationsNotProcessed(pageSize);
+        //TODO: Once we create a way of handling channel events in parallel, we can remove the MAX_NUMBER_PAGES_PROCESSED.
+        while (!CollectionUtils.isEmpty(pageOfAlertNotificationModels.getModels()) && numPagesProcessed < MAX_NUMBER_PAGES_PROCESSED) {
```

Comment:
Why are we getting a paged result and verifying that we process a specific separate number? Wouldn't we want to get the paged result and simply process all of them?

Reply:
There is a possibility where a customer can have a constant stream of notifications. We have a limitation right now Paulo and I discovered getting addressed in IALERT-2116. Right now, we must wait until it finishes any pages that are in the queue before it actually sends the event to the channels. Because of this we are putting in a hard upper limit so that at some point, even if it takes a long time, it will still send the messages to the channels.

Reply:
I guess my next question is the page is 100 items and the max number is 100. Why are these the same?

Reply:
Admittedly the numbers are fairly arbitrary, this setup would allow us to do 10,000 notifications. So each page can have up to 100 notifications, and we would process 100 pages before we would kick out of the loop and allow the notifications to be sent to the channels. With the setup once the next notification is picked up by the accumulator, the receiver would wake up again and continue processing where it left off.